### PR TITLE
Fix many deprecations warnings

### PR DESCRIPTION
**Pull request by abelsiqueira**
Wait for https://github.com/JuliaSmoothOptimizers/LinearOperators.jl/issues/30

### CODE REVIEW

File: src/nlp_types.jl [hunk 1]
```diff
@@ -140,22 +140,22 @@ immutable NLPModelMeta <: AbstractNLPModelMeta
...
     @rangecheck 1 ncon lin nln nnet lnet
 
-    ifix  = find(lvar .== uvar);
-    ilow  = find((lvar .> -Inf) & (uvar .== Inf));
-    iupp  = find((lvar .== -Inf) & (uvar .< Inf));
-    irng  = find((lvar .> -Inf) & (uvar .< Inf) & (lvar .< uvar));
-    ifree = find((lvar .== -Inf) & (uvar .== Inf));
-    iinf  = find(lvar .> uvar);
+    ifix  = find(lvar .== uvar)
+    ilow  = find(broadcast(&, lvar .> -Inf, uvar .== Inf))
```
Comment (dpo) [inline src/nlp_types.jl]: Is this really how it must be written? Is there no `&.`?

Comment (abelsiqueira) [inline src/nlp_types.jl]: You're right, I was using an old Compat.

**Comment by abelsiqueira** [comment]
LinearOperators 0.3.0 is not required to pass, only to decrease warning.

File: src/nlp_types.jl [hunk 2]
```diff
@@ -140,22 +140,22 @@ immutable NLPModelMeta <: AbstractNLPModelMeta
...
     @rangecheck 1 ncon lin nln nnet lnet
 
-    ifix  = find(lvar .== uvar);
-    ilow  = find((lvar .> -Inf) & (uvar .== Inf));
-    iupp  = find((lvar .== -Inf) & (uvar .< Inf));
-    irng  = find((lvar .> -Inf) & (uvar .< Inf) & (lvar .< uvar));
-    ifree = find((lvar .== -Inf) & (uvar .== Inf));
-    iinf  = find(lvar .> uvar);
+    ifix  = find(lvar .== uvar)
+    ilow  = find((lvar .> -Inf) .& (uvar .== Inf))
```
Comment (dpo) [inline src/nlp_types.jl]: Nice!!!

**Comment by dpo** [comment]
Many thanks!

**Pull request merged**